### PR TITLE
Add package documentation.

### PR DESCRIPTION
**Pull request by frankban**
Rendered: https://github.com/frankban/shapeup/blob/f0127b762b3feff1cacfdc45b1a17e0feceb79ce/README.md

### CODE REVIEW

**Review by jcsackett — COMMENTED** [review]
LGTM, thanks @frankban

**Review by jcsackett — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,4 +1,285 @@
...
+# shapeup
 
-This library provides the shape and reshape React property types, including
-the ability to declare frozen shapes.
+The React `shape` [property type](https://www.npmjs.com/package/prop-types) is
+useful for declaring how objects provided to components should look like. By
+stating that a property has a given shape, we are establishing a contract that
+must be fulfilled when instantiating the component. However, prop-types' shapes
+are to be considered more as a minimum requirement than as a proper external
+interface declaration: providing an object that has a superset of the declared
```
Comment (mitechie) [inline README.md]: However, prop-types' shapes are more a minimum requirement than as an external interface declaration

File: README.md [hunk 2]
```diff
@@ -1,4 +1,285 @@
...
+properties is not considered an error, and therefore components can end up
+relying on fields that are not part of the contract.
+
+The shapeup library fixes this by requiring the provided properties to only
+have the declared set of fields: when using shapeup, passing in an object with
```
Comment (mitechie) [inline README.md]: The shapeup library fixes this by forcing the provided properties to only be from the declared set of fields

Comment (mitechie) [reply README.md]: Why the : in the sentence? Should they just be periods. It's here and in the previous paragraph as well.

File: README.md [hunk 3]
```diff
@@ -1,4 +1,285 @@
...
+extraneous fields would result in an error. This way we ensure that what's
+declared in the contract is also what's actually used by the components.
```
Comment (mitechie) [inline README.md]: This way we ensure that what is declared in the contract exactly matches the components.

File: README.md [hunk 4]
```diff
@@ -1,4 +1,285 @@
...
+with the required fields, for instance:
+```javascript
+<MyComponent
+  api={{
+    getById: someobj.getById.bind(someobj),
+    getAll: someobj.getAll.bind(someobj)
+  }}
+/>
+```
+Anyway, in cases like this, in which there is `someobj` which already
```
Comment (mitechie) [inline README.md]: Cut "anyway"

File: README.md [hunk 5]
```diff
@@ -1,4 +1,285 @@
...
+rather than updating how every single component in the tree is instantiated.
+
+## Frozen shapes
+
+As with the traditional prop-types, it is possible to chain
+[`shapeup.shape`](#shape) with `isRequired`, in order to make that property
+required. It is also possible to chain the declaration with `frozen` in order
+to ensure that the provided property is deeply frozen. This is useful when
+providing a shape and willing to avoid the usual problems of object mutation,
+like unwanted side effects, bugs that are difficult to track down, unrequired
```
Comment (mitechie) [inline README.md]: providing a shape and wanting to avoid the usual problems of object mutation, such as unwanted side effects...

**Review by mitechie — COMMENTED** [review]
👍

File: README.md (hunk 1 above)
Comment (frankban) [reply README.md]: Done.

File: README.md (hunk 2 above)
Comment (frankban) [reply README.md]: Done.

File: README.md (hunk 3 above)
Comment (frankban) [reply README.md]: Done.

File: README.md (hunk 4 above)
Comment (frankban) [reply README.md]: Done.

File: README.md (hunk 5 above)
Comment (frankban) [reply README.md]: Done.

**Comment by frankban** [comment]
Thanks for the reviews!

**Pull request merged**